### ISSUE TITLE
How do you create the visualization of embeddings in hyperbolic space: wn-nouns.jpg?

### ISSUE THREAD

**Issue by mikalyoung** [facebookresearch/poincare-embeddings#4]

**Comment by Helw150** [comment]
@mnick I am also curious about visualization methods for these embeddings. Is there anything special to take into account when visualizing embeddings from hyperbolic space?

Seeing your visualization code would of course be helpful, but I'm assuming there is a consideration that keeps you from posting that in this repo.

**Comment by alex-bloom** [comment]
same question here...
what technique was used for mapping embeddings from, say, 5-dimensional Poincare ball (mammals) to 2-dimensional Euclidian ring for visualization? I tried SVD and while latent hierarchies survive to some extent after dimensionality reduction, the resulting plot is not that sharp as the one in the paper

**Comment by alex-bloom** [comment]
On the second thought, we all seem to be looking in a dark room for a black cat that is not there.
Taking a closer look at the paper, it says explicitly 
"Figure 2 shows a visualization of a **two-dimensional** Poincaré embedding." 
And again, in the Figure 2 capture: 
"Figure 2: **Two-dimensional** Poincaré embeddings of transitive closure of the WORDNET mammals
subtree... A Poincaré embedding with **d = 5** achieves mean rank 1.26 and MAP 0.927 on this subtree" - apparently, to make it clear that the Figure 2 does not represent the metrics obtained with the proper number of dimensions of the Poincaré ball

**Comment by lematt1991** [comment]
Yes, sorry for the late response, but @alex-bloom, you are correct.  The figure is of an embedding that is trained in only 2 dimensions (no dimensionality reduction is done).

**Comment by rtatgit** [comment]
> Yes, sorry for the late response, but @alex-bloom, you are correct. The figure is of an embedding that is trained in only 2 dimensions (no dimensionality reduction is done).

Hi,
Can you tell me after how many epochs did you achieve the above mentioned metrics?

**Comment by lematt1991** [comment]
Which metrics are you referring to?

**Comment by rtatgit** [comment]
> Which metrics are you referring to?

A Poincaré embedding with d = 5 achieves mean rank 1.26 and MAP 0.927. 
These evaluation metrics is what I was referring to.

**Comment by lematt1991** [comment]
Please create a separate issue.  This issue is regarding the visualization provided in the README

**Comment by joelkuiper** [comment]
Just wondering if there was any update on sharing the methods or code that generated the picture in README!

**Comment by adiell** [comment]
+1

**Comment by Veronicium** [comment]
Same here!

**Comment by jayachaturvedi** [comment]
Same here, would like to know how to generate the picture please. Thanks!
